packages/connect/src/api/cardanoGetPublicKey.ts

```diff
@@ -87,6 +90,9 @@ export default class CardanoGetPublicKey extends AbstractMethod<
     }
 
     async noBackupConfirmation() {
+        if (this.params.some(batch => batch.suppressBackupWarning)) {
```

Comment:
we only want to suppress if `show_display` === false **and** `suppressBackupWarning` === true, no?
and maybe only for trusted implementations? :)
 
I would do it like this:

in core.ts:
```
 await method.noBackupConfirmation(trustedHost && !isUsingPopup);
```
 
in each method:
```
 noBackupConfirmation(allowSuppression: boolean)
```
 
 in each params validation (method init):
 ```
 const show_display = typeof batch.showOnTrezor === 'boolean' ? batch.showOnTrezor : false;
 suppress_backup_warning: !show_display && batch.suppressBackupWarning,
 
 // or if we decide to have it as CommonParam then:
  suppress_backup_warning: !show_display && params.suppressBackupWarning,
 ```
 
 final condition:
 if it's trusted/nonpopup mode and all items in params bundle have suppressBackupWarning === true
 ```
if (allowSuppression && !this.params.some(batch => !batch.suppressBackupWarning)) {
    return true;
}
 ```
 
 

 if (allowSuppression && !this.params.some(p => p.show_display))

Reply:
sounds reasonable, but let me please think about it on Monday a bit. energy low now.

Reply:
> we only want to suppress if `show_display` === false **and** `suppressBackupWarning` === true, no? and maybe only for trusted implementations? :)

`showOnTrezor` param is not present on `getAccountInfo` method, so this would only apply for `getPublicKey` and `cardanoGetPublicKey`. In that case, `suppressBackupWarning` param is redundant because there is no case when `showOnTrezor == true && suppressBackupWarning == true`. And I would still have to specify `suppressBackupWarning: false` for all `getAccountInfo` calls. Did I miss something?

Reply:
what szymon says makes sense to me. just I would do

```
 await method.noBackupConfirmation(!isUsingPopup);
```
instead of 

```
 await method.noBackupConfirmation(trustedHost && !isUsingPopup);
```
imho it does not really matter if we force emitting some event in non trusted mode since we don't have any power over how implementator handles this evnet anyway. On the other hand `!isUsingPopup` makes perfect sense here because we want to show "no backup" screen in popup.

as for the noBackupConfirmation method, I would maybe do something like 

```
 async noBackupConfirmation(allowSupression) {
        if (allowSuppression && !this.params.filter(batch => batch.show_display).every((batch) => batch.suppressBackupWarning)) {
            return true;
        }
        
```
maybe every bundle item that has show_display:false should be also supressed?

and lastly, I would not include supression param into commonParams since we are working with single batch items in what I outlined here.

Reply:
> !this.params.filter(batch => batch.show_display).every((batch) => batch.suppressBackupWarning)

The first `!` shouldn't be there, right @mroz22?

> maybe every bundle item that has show_display:false should be also supressed?

In that case it could be just:

```ts
if (
    allowSuppression &&
    this.params.every(batch => batch.suppressBackupWarning || !batch.show_display)
) {
    return true;
}
```